src/core/CCInt.ml

```diff
@@ -37,6 +37,78 @@ let pow a b =
   pow 0 1 = 0
 *)
 
+let floor_div a n =
+  if a < 0 && n >= 0 then
+    (a + 1) / n - 1
+  else if a > 0 && n < 0 then
+    (a - 1) / n - 1
+  else
+    a / n
+
+(*$T
+  (floor_div 3 5 = 0)
```

Comment:
I'd be more comfortable with more tests with exact rounding (e.g. `floor_div (-20) 5 = -4` or sth like that)